jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java

```diff
@@ -65,29 +75,74 @@
      * When an entry can't be found in the cache, the global list is iterated
      * normally so the cache has no visible effect besides performance.
      */
-    private final ThreadLocal<List<Entry>> cache;
+
     private final Locker locker = new Locker();
     private final int maxEntries;
-    private final int cacheSize;
     private final AtomicInteger pending = new AtomicInteger();
+    private final Strategy strategy;
+    private final ThreadLocal<Entry> cache;
+    private final AtomicInteger next;
     private volatile boolean closed;
     private volatile int maxMultiplex = 1;
     private volatile int maxUsageCount = -1;
 
+    public enum Strategy
+    {
+        /**
+         * The Linear strategy looks for an entry always starting from the first entry.
+         * It will favour the early entries in the pool, but may contend on them more.
+         */
+        LINEAR,
```

Comment:
Since all the other names of this enum describe the starting index, I'd rename this one to something like `INDEX_ZERO`.

Reply:
How about `FIRST`?